src/components/App.tsx

```diff
@@ -1436,30 +1416,30 @@ export class App extends React.Component<any, AppState> {
       this.setState({ selectedElementIds: {} });
     }
 
-    if (isTextElement(element)) {
+    if (this.state.elementType === "text") {
       // if we're currently still editing text, clicking outside
       //  should only finalize it, not create another (irrespective
       //  of state.elementLocked)
       if (this.state.editingElement?.type === "text") {
         return;
       }
-      if (elementIsAddedToSelection) {
-        element = hitElement!;
-      }
```

Comment:
Removed this since I couldn't figure out what it does.

Reply:
I've tried a few things related to selection but they are all working. Not sure either

Reply:
The original purpose for this was to allow you to use the text tool to edit a text element. It doesn’t work now though, and I failed to comment it. I still think this is good functionality.

Reply:
It doesn't work on prod (that's what you meant?). So it probably needs rewriting. I'll create the issue for it, and keep this removed for this PR.